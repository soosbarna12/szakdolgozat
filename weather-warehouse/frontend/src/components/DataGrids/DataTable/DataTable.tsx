import { Box, Skeleton } from "@mui/material";
import { GridColDef } from "@mui/x-data-grid";
import React from "react";
import { StyledDataGrid } from "../../../stlyes/content.style";
import { HistoricalDataTable } from "../../../types/historicalDataTable.type";
import { getRowId } from "../../../utils/getRowId";


export function DataTable({ data, columns }: { data: HistoricalDataTable[], columns: GridColDef[] }) {
  function renderContent() {
    if (!data || data.length === 0) {
      return (
        <Skeleton
          variant="rectangular"
          animation="wave"
          width="100%"
          height={367}
          data-testid="dataTableSkeleton"
          sx={{ borderRadius: '10px' }}
        />
      );
    }
    return (
      <Box sx={{ width: "100%", height: 400 }} data-testid="dataTableContainer">
        <StyledDataGrid
          rows={data}
          getRowId={getRowId}
          columns={columns}
          hideFooterPagination
          hideFooterSelectedRowCount
          data-testid="dataTable"
        />
      </Box>
    )
  }

  return renderContent();
}
