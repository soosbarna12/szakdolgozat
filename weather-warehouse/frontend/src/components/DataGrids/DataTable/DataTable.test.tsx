import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import React from 'react';
import { renderWithQueryClient } from '../../../utils/test/renderWithQueryClient';
import { DataTable } from './DataTable';

jest.mock('react-leaflet', () => ({
  MapContainer: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="dataMapContainer">{children}</div>
  ),
  TileLayer: () => <div data-testid="mockTileLayer">Mock TileLayer</div>,
}));

jest.mock('@mui/x-data-grid', () => ({
  ...jest.requireActual('@mui/x-data-grid'),
  DataGrid: ({ rows }: { rows: any[]; columns: any[] }) => (
    <div data-testid="dataTable">
      {rows.length > 0 ? (
        rows.map((row, index) => <div key={index}>{JSON.stringify(row)}</div>)
      ) : (
        <div>No rows</div>
      )}
    </div>
  ),
}));

describe('DataGrids/DataTable', () => {
  const mockData = [
    { date: '2023-01-01', cityName: 'City1', countryCode: 'US' },
    { date: '2023-01-02', cityName: 'City2', countryCode: 'CA' },
  ];

  const mockColumns = [
    { field: 'date', headerName: 'Date', width: 150 },
    { field: 'cityName', headerName: 'City', width: 150 },
    { field: 'countryCode', headerName: 'Country', width: 150 },
  ];

  const renderComponent = (data = mockData, columns = mockColumns) => {
    return renderWithQueryClient(<DataTable data={data as any} columns={columns} />);
  };

  it('matches DataTable component snapshot', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders skeleton when data is empty', () => {
    renderComponent([]);
    const skeleton = screen.getByTestId('dataTableSkeleton');
    expect(skeleton).toBeInTheDocument();
  });

  it('renders data table with data', () => {
    renderComponent();
    const container = screen.getByTestId('dataTableContainer');
    const dataTable = screen.getByTestId('dataTable');

    expect(container).toBeInTheDocument();
    expect(dataTable).toBeInTheDocument();
  });

  it('renders data table with custom columns', () => {
    const customColumns = [
      { field: 'date', headerName: 'Date', width: 150 },
      { field: 'cityName', headerName: 'City', width: 150 },
    ];
    renderComponent(mockData, customColumns);

    const dataTable = screen.getByTestId('dataTable');
    expect(dataTable).toBeInTheDocument();
  });
});