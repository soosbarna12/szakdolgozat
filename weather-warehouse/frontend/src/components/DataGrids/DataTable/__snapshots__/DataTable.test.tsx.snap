// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrids/DataTable matches DataTable component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-2z7te8"
        data-testid="dataTableContainer"
      >
        <div
          data-testid="dataTable"
        >
          <div>
            {"date":"2023-01-01","cityName":"City1","countryCode":"US"}
          </div>
          <div>
            {"date":"2023-01-02","cityName":"City2","countryCode":"CA"}
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-2z7te8"
      data-testid="dataTableContainer"
    >
      <div
        data-testid="dataTable"
      >
        <div>
          {"date":"2023-01-01","cityName":"City1","countryCode":"US"}
        </div>
        <div>
          {"date":"2023-01-02","cityName":"City2","countryCode":"CA"}
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
